packages/node/src/node.ts

```diff
@@ -110,10 +113,13 @@ export class Node extends Base {
         const key = item[0];
         const value = item[1];
 
-        const result = this.sanitizeForEncoding(value);
+        const result = maxDepth > 0 ? this.sanitizeForEncoding(value, maxDepth-1) : undefined;
         if (result !== undefined){
           logClone[key] = result;
         }
+        if (CONTEXT_OBJECT_MAX_DEPTH_WARN && maxDepth === 0) {
+          console.warn(`Logtail context: max depth (${CONTEXT_OBJECT_MAX_DEPTH_DEFAULT}) reached. You must not embeds too deep or circular references.`);
```

Comment:
Just a slightly nicer copy:
```suggestion
          console.warn(`[Logtail] Max depth of ${CONTEXT_OBJECT_MAX_DEPTH_DEFAULT} reached when serializing logs. Please do not use circular references and excessive object depth in your logs.`);
```